### PR TITLE
Adding WHISPER support

### PR DESCRIPTION
**Pull request by awormuth**
Still need to write tests. Open to feedback for current changes!

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -169,8 +175,12 @@ func (c *Client) handleLine(line string) {
 			Tags:   message.Tags,
 			Text:   message.Text,
 		}
-
+		
 		switch message.Type {
+		case WHISPER:
```
Comment (gempir) [inline client.go]: Move this under the PRIVMSG, I prefer to sort these by occurence a bit and a PRIVMSG is going to be a lot more common usally.

File: message.go [hunk 2]
```diff
@@ -10,6 +10,8 @@ import (
 type msgType int
 
 const (
+	// Whisper
+        WHISPER msgType = 0
```
Comment (gempir) [inline message.go]: Just move this under ROOMSTATE like this:

```
const (
	// PRIVMSG standard chat message
	PRIVMSG msgType = iota + 1
	// CLEARCHAT timeout messages
	CLEARCHAT
	// ROOMSTATE changes like sub mode
	ROOMSTATE
	// WHISPER private user targeted whispers
	WHISPER
)
```

File: client.go [hunk 3]
```diff
@@ -41,6 +41,7 @@ type Client struct {
 	ircToken              string
 	connection            *tls.Conn
 	connActive            tAtomBool
+	onNewWhisper          func(channel string, user User, message Message)
```
Comment (gempir) [inline client.go]: channel makes no sense for whispers aswell, so just remove it.

**Comment by gempir** [comment]
Also an update for the README would be nice!

File: message.go (hunk 2 above)
Comment (awormuth) [reply message.go]: My concern with changing this is that message type ID for outgoing whisper is WHISPER is 0 (which is what I'm mainly using this for), and incoming whisper is 4 (not 100% confirmed)

File: client.go (hunk 1 above)
Comment (awormuth) [reply client.go]: Done!

File: client.go (hunk 3 above)
Comment (awormuth) [reply client.go]: Agreed. Good call.

**Comment by awormuth** [comment]
Any ideas regarding changes to message parsing? I'm not too familiar with the structure of the IRC messages and don't want to break existing logic.

File: message.go (hunk 2 above)
Comment (gempir) [reply message.go]: Okay. Hmm this was actually not intended to be used that way but I see why you are using it. It's okay to merge im gonna think how to refactor that part later

**Comment by gempir** [comment]
Like I said parseMiddle has a switch for msgType if you add whispers there it should already work.

**Comment by gempir** [comment]
@awormuth Hey. Checkout #18 I have merged your changes there and added some tests plus a whisper method.

**Pull request closed without merging**